setup.py

```diff
@@ -10,20 +10,15 @@
 
 install_requires: List[str] = []
 
-extras_require_test = [
-    "flake8",
-    "black",
-    "pytest-cov",
-    "pytest",
-]
+extras_require_test = ["flake8", "black", "pytest-cov", "pytest", "moto"]
```

Comment:
@emileten As we discussed a while back, I try to avoid `moto` in most of my libraries for 2 reasons.
1. It sometimes has the the effect of hiding service integration points and causes people to reduce separation of concerns in their code.  If I'm forced to explicitly mock and AWS service integration point when I write my code then it often leads to cleaner code where these service calls get encapsulated.
2. Explicit, limited surface area mocks make it much easier for reviewers to reason about what exactly the code is doing.

Reply:
Ideally we should just go ahead and write the mocks.  Most of the mocking you'll need is already available in https://github.com/developmentseed/stactools-pipelines/blob/main/stactools_pipelines/pipelines/conftest.py.  But I do think we need to improve the documentation for pipeline creators to show how to use these beyond just saying "Look at the example Sentinel-1 pipeline" so we should definitely explain this a bit better.

Reply:
I reverted this.

>  Ideally we should just go ahead and write the mocks 

Like I said this is really time consuming. It will be up to the user to write or not write these tests for his/her pipeline.